### PR TITLE
Add linkcheck step to be run once a week

### LINKED ISSUE #328: Automatically check for broken links
We need an automated link checker to check for 404 broken links in the documentaion.

Ideally it would be run during the CI checks.

### LINKED ISSUE #439: Add broken link check to CI for documentation builds
## 📚 Documentation
cc @kne42 

Proposal: add a linkchecker step to our [CI for documentation builds](https://github.com/napari/napari/blob/master/.github/workflows/make_docs.yml). The docs build should fail if it finds broken links.

I think this python package would do the job well, it's super simple to use: https://github.com/linkchecker/linkchecker

Reason for change: there are five broken links right now on the [napari.org](https://napari.org) main page alone. Lots more on other pages too. (Often we see this because the structure of the napari repo has changed, or a file is renamed slightly. There's less evidence of link rot from external sites we point to.)

### PR DESCRIPTION
**Pull request by melissawm**
# References and relevant issues
Closes #439 
Closes #328
Depends on #484 

# Description
Adds a sphinx linkcheck step to be run once a week. The reported failures will show up as issues in the repo. 

This check can also be triggered manually. I am using the sphinx linkchecker as it will check all links in the docs, including those coming from docstrings.

You can see this in action on my fork, here: 
https://github.com/melissawm/napari-docs/issues/16

### CODE REVIEW

**Comment by melissawm** [comment]
(I will get started on fixing the links currently flagged by the action)

**Comment by melissawm** [comment]
Hi all, any feedback here?

**Comment by psobolewskiPhD** [comment]
Shoot sorry to overlook this. I think link checking is a great idea. 
Nothing worse than hitting a broken link while looking for help.
This PR doesn't really affect anything in the repo per se, since it just opens issues, so I'm going to approve and suggest we optimistically merge and then iterate?

**Review by psobolewskiPhD — APPROVED** [review]

**Comment by melissawm** [comment]
Let me submit a PR to fix the existing failures first so we don't get spammed by the failures :D

File: .github/workflows/linkcheck.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          python -m pip install --upgrade pip
+          python -m pip install "napari/[all]"
+        env:
+          PIP_CONSTRAINT: ${{ github.workspace }}/napari/resources/constraints/constraints_py3.10_docs.txt
+
+      - name: Run link checker
+        uses: aganders3/headless-gui@v2
+        env:
+          GOOGLE_CALENDAR_ID: ${{ secrets.GOOGLE_CALENDAR_ID }}
+          GOOGLE_CALENDAR_API_KEY: ${{ secrets.GOOGLE_CALENDAR_API_KEY }}
```
Comment (willingc) [inline .github/workflows/linkcheck.yml]: Why does this action need `GOOGLE_CALENDAR` secrets?

**Review by willingc — APPROVED** [review]
LGTM. 1 question for my understanding. TYVM.

File: .github/workflows/linkcheck.yml (hunk 1 above)
Comment (melissawm) [reply .github/workflows/linkcheck.yml]: Maybe we don't 🤔 This is here so we can build the docs with the google calendar embedded but we don't need that for link checking. I'll remove

**Pull request merged**